admin/plugins.php

```diff
@@ -390,3 +426,82 @@ function perflab_render_plugin_card( array $plugin_data ) {
 	</div>
 	<?php
 }
+
+// WordPress AJAX action to handle the button click event.
+add_action( 'wp_ajax_perflab_install_activate_standalone_plugins', 'perflab_install_activate_standalone_plugins_callback' );
+
+/**
+ * Handles the standalone plugin install and activation via AJAX.
+ *
+ * @since n.e.x.t
+ */
+function perflab_install_activate_standalone_plugins_callback() {
+	if ( ! wp_verify_nonce( $_REQUEST['nonce'], 'perflab-install-activate-plugins' ) ) {
+		$status['errorMessage'] = esc_html__( 'Invalid nonce: Please refresh and try again.', 'performance-lab' );
+		wp_send_json_error( $status );
+	}
+
```

Comment:
Here we'll need to also include a capability check to make sure that the current user can both install and activate plugins.

Reply:
@felixarntz We already checked it on JS code. I don't think we needs to check it again. WDYT?

```
if ( ! perflab_module_migration_notice.has_permission ) {
   alert( perflab_module_migration_notice.permission_error );
   return;
}
```

Reply:
@mukeshpanchal27 Checking permissions on the client isn't safe. It can be useful for sure, but anybody could make a request to the server manually. So it has to always be checked on the server.

On the client-side it's a nice to have. I'd argue you could remove it from the JS code to make it a bit simpler.

Reply:
Got it. Updated in https://github.com/WordPress/performance/pull/899/commits/abfc2f3b1976b24e987a1ef3dcfa6fcc59068ce5